stdlib/public/SDK/Foundation/JSONEncoder.swift

```diff
@@ -2359,11 +2401,34 @@ extension _JSONDecoder {
         }
     }
 
+    fileprivate func unbox<T>(_ value: Any, as type: _JSONStringDictionaryDecodableMarker.Type) throws -> T? {
+        var result = [String : Any]()
+        guard let dict = value as? NSDictionary else {
+            throw DecodingError._typeMismatch(at: self.codingPath, expectation: type, reality: value)
+        }
+        let elementType = type.elementType
+        for (k, v) in dict {
+            guard let key = k as? String else {
```

Comment:
You can `as!` here; `[NS]JSONSerialization` won't return a dictionary with anything but string keys